core/utils/drag.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as browserEvents from '../browser_events.js';
+import type {WorkspaceSvg} from '../workspace_svg.js';
+import {Coordinate} from './coordinate.js';
+
+const workspaceToDragDelta: WeakMap<WorkspaceSvg, Coordinate> = new WeakMap();
+
+/**
+ * Convert from mouse coordinates to workspace coordinates.
+ *
+ * @param workspace The workspace where the pointer event is occurring.
+ * @param e The pointer event with the source coordinates.
+ */
+function mouseToWorkspacePoint(
+  workspace: WorkspaceSvg,
+  e: PointerEvent,
+): SVGPoint {
+  const point = browserEvents.mouseToSvg(
+    e,
+    workspace.getParentSvg(),
+    workspace.getInverseScreenCTM(),
+  );
+  // Fix scale of mouse event.
+  point.x /= workspace.scale;
+  point.y /= workspace.scale;
```

Comment:
Since I don't really know what a `ScreenCTM` is the following question is probably nonsense, but: can the post-`mouseToSvg` fixup be avoided by passing a better value as the third parameter of `mouseToSvg`?

And if so would such a change be beneficial?

(Side note: looking at the definitions of `mouseToSvg` and `getInverseScreenCTM`, it appears that it mightalternatively be possible to completely omit the third parameter to `mouseToSvg` with no change to the returned valuest—but perhaps that would be less efficient, since `getInverseScreenCTM` seems to do caching.)

Reply:
I think, in some sense, the "correct" thing to do would be to pass workspace.getCanvas() as the second argument, to derive the transform matrix from the block canvas which should incorporate the zoom. However, the SVG APIs that mouseToSvg uses require an SVGSVGElement (`<svg>`), whereas the block canvas is a `<g>` element.